server/service/devices.go

```diff
@@ -54,9 +55,15 @@ func getDeviceHostEndpoint(ctx context.Context, request interface{}, svc fleet.S
 		return getDeviceHostResponse{Err: err}, nil
 	}
 
+	license, err := svc.License(ctx)
+	if err != nil {
+		return nil, err
```

Comment:
`return getDeviceHostResponse{Err: err}, nil`

Reply:
@lucasmrod I'm planning to do a quick PR to fix this and follow up on the other two comments separately